### PR TITLE
SOLR-17357 Improve default --help formatting for CLI Tools (branch_9x)

### PR DESCRIPTION
**Pull request by janhoy**
This will be merged AFTER #2540, as it is backported on top of those changes. It will then replace some of the new code in that PR with what was committed to main.

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/cli/SolrCLI.java
Comment (epugh) [inline solr/core/src/java/org/apache/solr/cli/SolrCLI.java:192]: nice to get rid of this odd code!

**Review by epugh — APPROVED** [review]
LGTM!

Would be nice to get this into 9.7, and then hopefully that will set us up for smoother backports from `main` in future releases...

**Comment by epugh** [comment]
I want to click the Squash and Merge button, but don't totally understand the target branch, so will hold off!

File: .github/workflows/bin-solr-test.yml
Comment (janhoy) [inline .github/workflows/bin-solr-test.yml:12]: @epugh I added the change I proposed in https://github.com/apache/solr/pull/2573#pullrequestreview-2188208038 to trigger bats tests for all changes in cli code.

NOTE this is not in main, so perhaps add it in another PR?

Comment (janhoy) [reply .github/workflows/bin-solr-test.yml]: Never mind, reverted.

**Comment by janhoy** [comment]
> I want to click the Squash and Merge button, but don't totally understand the target branch, so will hold off!

Moved it to correct branch, thik it is ready. Was expecting bats tests to run after merging #2575 but they did not. Guess we'll need to do some (but not all) bats test adjustments as well for the back port. Not all, since main uses more help output from SolrCLI than branch_9x...

**Comment by epugh** [comment]
Fingers crossed a good build!   Changes look good.

**Pull request merged**
